addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,345 @@
+                        updateState(CHANNEL_POWER, OnOffType.OFF);
+                        power = false;
+                    }
+                } else if ("dimmer".equals(command)) {
+                    updateState(CHANNEL_BRIGHTNESS, readDimmer());
+                } else if ("freq".equals(command)) {
+                    updateState(CHANNEL_FREQUENCY, readFrequency());
+                } else if ("source".equals(command)) {
+                    updateState(CHANNEL_SOURCE, new StringType(readUntil('!')));
+                } else if ("display".equals(command)) {
+                    String stringLength = readUntil(',');
+                    int length = Integer.parseInt(stringLength);
+                    byte[] data = new byte[length];
+                    int off = 0;
+                    while (off != length) {
+                        int r = serialPort.getInputStream().read(data, off, length - off);
+                        if (r == -1) {
+                            throw new IOException("Connection closed while reading display content");
+                        } else {
+                            off += r;
+                        }
+                    }
+                    // We don't do anything with display content, could add a channel if this is useful
+                } else {
+                    readUntil('!'); // discard
+                }
+
+            } catch (IOException e) {
+                if (serialPort != null) {
+                    logger.debug("Input error while receiving data from amplifier, waiting...", e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                    disconnect();
+                    try {
+                        Thread.sleep(ERROR_RETRY_DELAY_MS);
+                    } catch (InterruptedException e1) {
+                        return;
+                    }
+                }
+            } catch (Exception e) {
+                if (serialPort != null) { // If serial port is closed, it's set to null,
+                                          // there is no message here,
+                    logger.warn("Unexpected error", e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                            "Unknown error while processing input: " + e.getMessage());
+                    disconnect();
+                    try {
+                        Thread.sleep(ERROR_RETRY_DELAY_MS);
+                    } catch (InterruptedException e1) {
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    private void send(String text) throws IOException, ConfigurationError {
+        if (power) {
+            connect();
+            serialPort.getOutputStream().write(text.getBytes(StandardCharsets.US_ASCII));
+        }
+    }
+
+    private void sendForce(String text) throws IOException, ConfigurationError {
```

Comment:
This is a personal opinion: I would rename `sendForce` to `send` and `send` to `sendWhenPowerOn`. This because there is nothing "forced" it will just send. While your "send" does the send conditionally. So to better reflect what is actually happening i would recommend the renaming.

Reply:
Also I think your `send` should call `sendForce` with the if condition to simplify code.